project/profiles/capacity4more/themes/c4m/kapablo/build/src/javascripts/kapablo.js

```diff
@@ -700,29 +716,31 @@ var jQuery = jQuery || {};
       this.emptyWidgetFields = emptyFields;
     },
 
-    checkTopicFields: function () {
+    checkAngularFields: function () {
       var emptyFields = false;
-      this.requiredTopicFields.each(function () {
+      this.requiredAngularFields.each(function () {
         if ($(this).find('.selected-values > .ng-scope:not(.ng-hide)').length === 0) {
           emptyFields = true;
         }
       });
 
-      this.emptyTopicFields = emptyFields;
+      this.emptyAngularFields = emptyFields;
     },
 
     checkFields: function () {
-      this.checkWidgetFields();
       this.checkImageFields();
+      this.checkDragAndDropFields();
       this.checkTextFields();
-      this.checkTopicFields();
+      this.checkWidgetFields();
+      this.checkAngularFields();
     },
 
     initializeFields: function () {
-      this.requiredTextFields = this.forms.find('.required');
       this.requiredImageFields = this.forms.find('.field-type-image').has('.form-required');
+      this.requiredDragAndDropFields = this.forms.find('.field-widget-dragndrop-upload-file').has('.form-required');
+      this.requiredTextFields = this.forms.find('.required');
       this.requiredWidgetFields = this.forms.find('.required-checkbox');
-      this.requiredTopicFields = this.forms.find('.c4m_vocab_topic').has('.form-required');
+      this.requiredAngularFields = this.forms.forms.find('.c4m_vocab_topic, .c4m_vocab_document_type').has('.form-required');
```

Comment:
Think there's duplication here? Gives an error in console that blocks the page from rendering
```TypeError: undefined is not an object (evaluating 'this.forms.forms.find')```